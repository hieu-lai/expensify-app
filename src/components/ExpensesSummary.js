import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
  <div>
    <h3>
      Viewing {props.expenses.length} {
        props.expenses.length === 1 ? 'expense' : 'expenses'  
      } totalling {
        numeral(selectExpenseTotal(props.expenses) / 100).format('$0,0.00')
      }
    </h3>
  </div>
);

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)  
});

export default connect(mapStateToProps)(ExpensesSummary);