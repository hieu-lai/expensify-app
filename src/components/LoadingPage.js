import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <img className="loader__image" src="/images/loader.gif" alt="Loadding..." />
  </div>
);

export default LoadingPage;