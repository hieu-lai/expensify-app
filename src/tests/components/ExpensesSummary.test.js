import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render 2 expenses correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render 1 expense correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
  expect(wrapper).toMatchSnapshot();
});