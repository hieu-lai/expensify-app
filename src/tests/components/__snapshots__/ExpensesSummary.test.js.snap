// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense correctly 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h3>
</div>
`;

exports[`should render 2 expenses correctly 1`] = `
<div>
  <h3>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
  </h3>
</div>
`;
